plenum/server/consensus/view_change_service.py

```diff
@@ -301,7 +301,11 @@ def _finish_view_change(self):
         self.last_completed_view_no = self._data.view_no
 
     def _propose_view_change(self, suspision_code):
-        proposed_view_no = self._data.view_no + 1
+        proposed_view_no = self._data.view_no
+        # TODO: For some reason not incrementing view_no in most cases leads to lots of failing/flaky tests
```

Comment:
I didn't get this TODO comment... The only situation when we don't need to increment viewNo is `STATE_SIGS_ARE_NOT_UPDATED ` code. So, what is the issue?

Reply:
It was ported from original code in old ViewChanger. Some time ago (last year AFAIR) we had a theory that during view change we shouldn't send IC with future view no unless it is a view change timeout itself. For some reason that didn't work, at that time we gave up and decided to just leave a comment for a better time. I'm just preserving it.